hayeon/programmers-1/행렬의-덧셈.js

```diff
@@ -0,0 +1,48 @@
+function solution(arr1, arr2) {
+  let result = [[]];
+  for (let i = 0; i < arr1.length; i++) {
+    result[i] = [];
+    for (let j = 0; j < arr1[i].length; j++) {
+      result[i].push(arr1[i][j] + arr2[i][j]);
+    }
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(arr1, arr2) {
  const result = [];
  
  for (let i = 0; i < arr1.length; i++) {
    const temp = [];
    
    for (let j = 0; j < arr1[i].length; j++) {
      temp.push(arr1[i][j] + arr2[i][j]);
    }
    
    result.push(temp);
  }
  return result;
}
```
- result를 2중배열로 선언해서 i를 이용한 방법도 좋았습니다 👍 그런데 2번째 풀이처럼 for문 안에다 sum이라는 배열을 활용하는 코드가 더 직관적이고 자주 활용되는 방법입니다. 참고로 저는 sum은 숫자를 더한 `값`이라 배열에는 부적합하다고 생각이 드네요 sumArr이나, temp와 같은 변수명이 더 적절한 것 같습니다 🙇‍♂️
- 아래는 2중 map을 이용한 풀이입니다 참고해주시면 감사드려요~
```js
function solution(arr1, arr2) {
  return arr1.map((innerArr, i) => {
    return innerArr.map((el, j) => el + arr2[i][j]);
  });
}
```

Reply:
변수명을 더 고려해서 만들어보겠습니다! 감사합니다 ㅎㅎ🧐